### PR TITLE
Issue#857: MyRocks: Incorrect behavior when multiple statements fail …

### PR DESCRIPTION
**Pull request by spetrunia**
…inside a transaction

rollback_to_stmt_savepoint() calls do_rollback_to_savepoint(). This
removes the changes made by this statement, and also removes the last
set savepoint.

Before we start processing the next statement, we need to set the new
savepoint, so we will have something to rollback to if the next statement
fails.

Since rollback_to_stmt_savepoint always sets a new savepoint now,
m_n_savepoints is now redundant and is removed.

### CODE REVIEW

File: mysql-test/suite/rocksdb/r/transaction.result [hunk 1]
```diff
@@ -958,3 +958,20 @@ a
 rollback;
 drop function func;
 drop table t1,t2,t3;
+#
+# MDEV-16710: Slave SQL: Could not execute Update_rows_v1 event with RocksDB and triggers
+# Issue#857: MyRocks: Incorrect behavior when miltiple statements fail inside a transaction
```
Comment (hermanlee) [inline mysql-test/suite/rocksdb/r/transaction.result]: nit: typo: multiple

File: storage/rocksdb/ha_rocksdb.cc [hunk 2]
```diff
@@ -2436,7 +2436,6 @@ class Rdb_transaction {
       entire transaction.
     */
     do_set_savepoint();
-    m_n_savepoints= 1;
```
Comment (hermanlee) [inline storage/rocksdb/ha_rocksdb.cc:2439]: Since this is not longer used, the definition in Rdb_transaction should also be removed.

**Review by hermanlee — CHANGES_REQUESTED** [review]
Can you also update the description with:

Squash with D7509380

On a new rebase, we'll merge this fix with the original change.

File: storage/rocksdb/ha_rocksdb.cc (hunk 2 above)
Comment (spetrunia) [reply storage/rocksdb/ha_rocksdb.cc:2439]: Done

**Comment by spetrunia** [comment]
Addressed all review input.

**Review by hermanlee — APPROVED** [review]

**Pull request closed without merging**